### PR TITLE
Secret creation should be using stringData

### PR DESCRIPTION
**Pull request by jhadvig**
Fixes https://github.com/openshift/origin-web-console/issues/2651

@spadgett PTAL

### CODE REVIEW

File: app/scripts/directives/createSecret.js [hunk 1]
```diff
@@ -117,38 +117,38 @@ angular.module("openshiftConsole")
             data: {}
```
Comment (spadgett) [inline app/scripts/directives/createSecret.js]: nit: It's a little simpler to change `data` here to `stringData` and just set `secret.stringData` in each case block below.

File: app/scripts/directives/createSecret.js [hunk 2]
```diff
@@ -117,38 +117,38 @@ angular.module("openshiftConsole")
...
           };
 
+          var secretData = {};
           switch (authType) {
             case "kubernetes.io/basic-auth":
               // If the password/token is not entered either .gitconfig or ca.crt has to be provided
               if (data.passwordToken) {
-                secret.data = {password: window.btoa(data.passwordToken)};
+                secretData['password'] = data.passwordToken;
```
Comment (spadgett) [inline app/scripts/directives/createSecret.js]: We typically don't use the bracket `[]` notation unless the key has special characters.

```js
secretData.password = data.passwordToken;
```

**Comment by jhadvig** [comment]
@spadgett comments addressed

**Pull request merged**